### PR TITLE
PKCS #11 Remove `iot` prefix.

### PR DESCRIPTION
**Pull request by lundinc2**
PKCS #11 Remove `iot` prefix.

Description
-----------
This change removes the `iot` prefix, and swaps it to the `core` prefix for PKCS files.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by karkhaz** [comment]
Hi! For some reason CI is failing to check out a submodule when we clone this code:
```
fatal: failed to unpack tree object 9e0a3c63943644bd1996884908d65dd117e75994
error: Submodule 'libraries/abstractions/pkcs11/corePKCS11' could not be updated.
error: Submodule 'libraries/abstractions/pkcs11/corePKCS11' cannot checkout new HEAD
```
this is when attempting to (recursively) check out 492401e that you pushed just now.

**Comment by lundinc2** [comment]
Hey @karkhaz is this because the PKCS repo doesn't have the git modules `update = none` option set?

**Comment by lundinc2** [comment]
> /bot run checks

I've aligned it with the others, not sure why this one is failing.

**Review by sarenameas — DISMISSED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -29,7 +29,7 @@
 	branch = mbedtls-2.16.8
 [submodule "libraries/abstractions/pkcs11/psa"]
 	path = libraries/abstractions/pkcs11/psa
-	url = https://github.com/Linaro/freertos-pkcs11-psa.git
+	url = https://github.com/lundinc2/freertos-pkcs11-psa.git
```
Comment (sarenameas) [inline .gitmodules:32]: Is this right? Do you want to change to your repo?

Comment (lundinc2) [reply .gitmodules:32]: No, this was a temporary change to test the regressions. I have a PR to upstream this, once that is merged, this is ready.

Comment (sarenameas) [reply .gitmodules:32]: Should this be in draft to keep CBMC from running?

**Comment by lundinc2** [comment]
@karkhaz FYI there are no cbmc proofs in the corePKCS11 repo, is the CI checking for some in there, and failing?

**Review by n9wxu — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Pull request merged**